core/src/main/java/jenkins/telemetry/impl/java11/MissingClassTelemetry.java

```diff
@@ -0,0 +1,367 @@
+
+import com.google.common.annotations.VisibleForTesting;
+import hudson.Extension;
+import io.jenkins.lib.versionnumber.JavaSpecificationVersion;
+import jenkins.model.Jenkins;
+import jenkins.telemetry.Telemetry;
+import jenkins.util.java.JavaUtils;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.TimeZone;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Telemetry class to gather information about missing classes when running on java 11. This class sends classes not
+ * found and in packages related with Java changes from Java 8 to Java 11. See {@link #MOVED_PACKAGES}.
+ **/
+
+@Extension
+@Restricted(NoExternalUse.class)
+public class MissingClassTelemetry extends Telemetry {
+    private static final Logger LOGGER = Logger.getLogger(MissingClassTelemetry.class.getName());
+
+    // Store 100 events today
+    private static MissingClassEvents events = new MissingClassEvents();
+
+    // When we begin to gather these data
+    private final static LocalDate START = LocalDate.of(2019, 4, 1);
+    // Gather for 2 years (who knows how long people will need to migrate to Java 11)
+    private final static LocalDate END = START.plusMonths(24);
+
+    // The types of exceptions which can be reported
+    private static final Set reportableExceptions =
+            new HashSet<Class>(Arrays.asList(ClassNotFoundException.class, NoClassDefFoundError.class));
+
+    @VisibleForTesting
+    /* package */ static final String CIRCULAR_REFERENCE = "Circular reference found on the exception we are analysing to report via telemetry";
+
+    /**
+     * Packages removed from java8 up to java11
+     * https://blog.codefx.org/java/java-11-migration-guide/
+     */
+    private final static String[] MOVED_PACKAGES = new String[] {"javax.activation", "javax.annotation", "javax.jws",
+            "javax.rmi", "javax.transaction", "javax.xml.bind", "javax.xml.soap", "javax.xml.ws", "org.omg",
+            "javax.activity", "com.sun", "sun"};
+
+    /**
+     * Places where a ClassNotFoundException is going to be thrown but it's ignored later in the code, so we
+     * don't have to send this exception, even though it might be related with java classes of moved packages
+     */
+    private static String[][] IGNORED_PLACES = {
+            {"hudson.util.XStream2$AssociatedConverterImpl", "findConverter"},
+            {"org.jenkinsci.plugins.workflow.cps.global.GrapeHack", "hack"},
```

Comment:
It references a code in a plugin,not in the Jenkins core. If we want to go forward with such approach, maybe we should introduce a way for plugins to define their own exceptions to be filtered.

Reply:
Moving it into plugin code imply making plugin authors get involved in this stuff, or doing harder the maintenance of this ignored places by having to do plugins releases. 

I think on another approach, it is doing the telemetry implementation to download the ignored places from a kind of json endpoint. With this approach, we can add ignored places easily, they are used without having to release a new Jenkins version or different plugin versions. But we can do that in the next iteration.